packages/zosjobs/src/ModifyJobs.ts

```diff
@@ -0,0 +1,108 @@
+        options: { jobclass?: string, hold?: boolean, release?: boolean }): Promise<undefined|IJobFeedback> {
+        this.log.trace("ModifyJob called with jobname %s jobid %s", parms);
+        return ModifyJobs.modifyJobCommon(session, parms, options);
+    }
+
+    /**
+     * Modify a job
+     * Full version of the API with a parameter object
+     * @static
+     * @param {AbstractSession} session - z/OSMF connection info
+     * @param {IModifyJobParms} parms - parm object (see IModifyJobParms interface for details)
+     * @param {IModifyJobOptions} options - options object (see IModifyJobOptions interface for details)
+     * @returns {Promise<undefined|IJobFeedback>} - promise of undefined, or IJobFeedback object returned by API if modifyVersion is 2.0
+     * @memberof ModifyJobs
+     */
+    public static async modifyJobCommon(
+        session: AbstractSession,
+        parms: IModifyJobParms,
+        options: IModifyJobOptions
+    ): Promise<undefined|IJobFeedback> {
+        this.log.trace("ModifyJobCommon called with parms %s", JSON.stringify(parms));
+        ImperativeExpect.keysToBeDefinedAndNonBlank(parms, ["jobid", "jobname"],
+            "You must specify both the jobname and jobid for the job you want to modify.");
+        this.log.info("Modifying job %s.%s", parms.jobname, parms.jobid);
+
+        const headers: any = [Headers.APPLICATION_JSON];
+        const parameters: string = "/" + parms.jobname + "/" + parms.jobid;
+        let response: IJobFeedback;
+        let request: IModifyJob;
+        let mergedMessage: string = "";
+        let exception: boolean = false;
+
+        if(options.hold && options.release){
+            throw new ImperativeError({msg: "Parameters `hold` and `release` are in conflict and cannot be specified together"});
+        }
+
+        if(options.hold || options.release){
+            options.hold ? request = { request: "hold"} : request = { request: "release"};
+            try{
+                response = await ZosmfRestClient.putExpectJSON(session, JobsConstants.RESOURCE + parameters, headers, request);
+            }
+            catch(err){
+                exception = true;
+                err.mMessage=err.mMessage.concat('Modification Error');
+                throw err;
+            }
+            mergedMessage = mergedMessage + '\n' + response.message;
+        }
+
+        // build request to change class, only if defined and no exception from potential previous request
+        if (options.jobclass != undefined && !exception){
+            request = {
+                class: options.jobclass,
+            };
+            response = await ZosmfRestClient.putExpectJSON(session, JobsConstants.RESOURCE + parameters, headers, request);
+            mergedMessage = mergedMessage + '\n' + response.message;
+        }
+        response.message = mergedMessage;
+        return response;
+    }
+
+
+    /**
+     * Getter for brightside logger
```

Comment:
This code shouldn't mention brightside. It should just be a logger, since any app could theoretically use Imperative and have its own app logger.